src/google/protobuf/compiler/command_line_interface.cc

```diff
@@ -188,8 +188,25 @@ bool TryCreateParentDirectory(const string& prefix, const string& filename) {
 // Get the absolute path of this protoc binary.
 bool GetProtocAbsolutePath(string* path) {
 #ifdef _WIN32
-  char buffer[MAX_PATH];
-  int len = GetModuleFileName(NULL, buffer, MAX_PATH);
+  wchar_t wide_buffer[MAX_PATH];
+  // Note: the returned length does not include terminating null character.
+  int len = GetModuleFileNameW(NULL, wide_buffer, MAX_PATH);
```

Comment:
Yes almost always.  Basically what encoding is used in GetModuleFileNameA depends on system locale. For example it uses Shift-JIS (Codepage: 932 to be precise) when the system locale is Japanese.

Basically GetModuleFileNameA exists for compatibility reasons to support applications that do not support Unicode at all. Depending on GetModuleFileNameA in newly implemented code doesn't make sense in almost all the situations.

![region_settings](https://cloud.githubusercontent.com/assets/5861397/8897375/2d7007fe-33c9-11e5-95cb-0aa3d1d7cebb.png)